src/components/CommentsList/CommentsList.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { Comment, CommentData } from '../../types';
+import {
+  deleteCommentById,
+  getCommentsByPostId,
+  postComment,
+} from '../../api';
+import { NewCommentForm } from '../NewCommentForm';
+
+type Props = {
+  postId: number,
+};
+
+export const CommentsList: React.FC<Props> = ({ postId }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getCommentsByPostId(postId)
+      .then(postComments => {
+        // setHasLoadedData(true);
+        setComments(postComments);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, [postId]);
+
+  useEffect(() => setIsFormVisible(false), [postId]);
+
+  const handleDeleteComment = (commentId: number) => {
+    const preDeleteComments = [...comments];
+
+    setComments(prev => (
+      prev.filter(comment => comment.id !== commentId)
+    ));
+    deleteCommentById(commentId)
+      .catch(() => {
+        setComments(preDeleteComments);
+      });
+  };
+
+  const handlePostComment = (commentData: CommentData) => {
+    const newComment = {
+      id: 0,
```

Comment:
do we need set id for the comment, that we need to add?
check description for `post` method in api